### PR TITLE
[WIP] add the debug toolbar, with its own env var

### PR DESCRIPTION
**Pull request by Pomax**
Related to https://github.com/mozilla/foundation.mozilla.org/issues/5495, adds a the django debug toolbar, enabled via an env var that is False by default. (because running with the debug toolbar slows things WAAAAAY down. It should not be turned on by default, not even for local dev)

⚠️⚠️⚠️ THIS PR IS TOO IMPRACTICAL TOO LAND ATM ⚠️⚠️⚠️

The debug toolbar's SQL investigation code makes pages take _forever_ if they run too many queries.

### CODE REVIEW

File: env.default [hunk 1]
```diff
@@ -18,14 +18,14 @@ CORS_ALLOWED_ORIGIN_REGEXES=
 DATABASE_URL=
 DEBUG=True
 VSCODE_DEBUGGER=False
-DISABLE_DEBUG_TOOLBAR=True
```
Comment (Pomax) [inline env.default:21]: left-over from years ago

File: requirements.in [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
 boto3
-Django==2.2.16
+cloudinary
 dj-database-url
-djangorestframework
+django_csp
 django-admin-sortable==2.2.3
 django-cors-headers
-django_csp
+django-debug-toolbar
```
Comment (Pomax) [inline requirements.in:8]: this is the only new thing - the rest is just from alphabetic sort

**Comment by KalobTaulien** [comment]
Hmm this took 147 seconds to render the homepage, and frankly there isn't much on the homepage. I got this down to 44 seconds by disabling a few things.

```python
    settings.DEBUG_TOOLBAR_CONFIG = {
        "SHOW_TOOLBAR_CALLBACK": use_toolbar,
        "RESULTS_CACHE_SIZE": 100,
        'PROFILER_MAX_DEPTH': 5,  # I added this 
    }

    # Everything below this I added to my settings 
    settings.DEBUG_TOOLBAR_PANELS = [
        'debug_toolbar.panels.timer.TimerPanel',
        'debug_toolbar.panels.settings.SettingsPanel',
        'debug_toolbar.panels.sql.SQLPanel',
        'debug_toolbar.panels.staticfiles.StaticFilesPanel',
        'debug_toolbar.panels.templates.TemplatesPanel',
        'debug_toolbar.panels.cache.CachePanel',
    ]
    settings.INTERNAL_IPS = [
        '127.0.0.1',
        '172.17.0.1',
    ]
```

I'm also thinking we should refactor this. Right now every environment build is going to download DDT but we really only need it on localhost. That's adding to our build time (and bandwidth if that's being measured). 

I'm leaving my idea here and we can take what's good and leave behind what's bad. What if:
1. We added this to the main settings.py file:
```python
try:
    from local_settings import *
except ImportError:
    pass
```
And provided a local_settings.py.example file for the repo that we could enable, and have DDT enable-able in there (off by default because it's so slow) with settings (like the ones above) that we can enable or disable locally to speed up profiling based on what we're looking for. 
2. Use a second requirements file that includes the current (main) requirements file but only use it for local dev - we could put local tools like django-extensions and django-debug-toolbar in there. This way prod will never have these tools on its server. 
3. Create a new `inv` command for local build with debugging tools. 

^ My thinking here is that we don't use these tools very often so they shouldn't be constantly downloaded or checked if they're installed.

**Comment by Pomax** [comment]
That sounds reasonable, although the separation of "requirements.in" vs. "requirements.txt" might make this harder than we want (I can't remember why we moved to .in files, so this might be a time to evaluate whether or not that move made sense)

**Review by KalobTaulien — CHANGES_REQUESTED** [review]
I think we should look at this a bit deeper so we at least dont need to have _every_ new environment download DDT if it's not being used. Left further comments above.

**Comment by Pomax** [comment]
just not relevant atm

**Pull request closed without merging**